### PR TITLE
Template Apache Spark containers for cross building (resolves #196)

### LINKED ISSUE #196: Cross build spark versions
It'd be good to build the Spark containers for:
- 1.5.2
- 1.6.2
- 2.0.1

@hannes-ucsc and I would like this for https://github.com/BD2KGenomics/cgcloud/issues/231

### PR DESCRIPTION
**Pull request by fnothaft**
Modifies apache-spark-{common,master,worker} for cross building to Spark 1.5.2, 1.6.2, and 2.0.1.

Haven't tested this locally yet; I'm on too wimpy of a connection to download all the Spark build artifacts and too wimpy of a box to build Spark thrice.

### CODE REVIEW

File: apache-spark-master/Makefile [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
 
-build: ${build_tool}
+build: $(build_tool)
 
-${build_tool}: Dockerfile run-master.sh
+%.spark: sparkver=$(@:.spark=)
+%.spark: Makefile.template Dockerfile.template
+	mkdir -p $@
+	cp run-master.sh ${@}/
+	sed -e "s:SPARKVER:$(sparkver):g" -e "s:COMMONCOMMIT:$(common_git_commit):g" Dockerfile.template > ${@}/Dockerfile
```
Comment (hannes-ucsc) [inline apache-spark-master/Makefile:13]: May I bring the Dockerfile.py approach that I used for the Toill appliance to the discussion table? I like that it keeps the logic all in one place. While `make` is Turing complete, I'd rather not wrestle with its peculiarities. And believe me, I have wrestled with it a lot in my life.

Comment (fnothaft) [reply apache-spark-master/Makefile:13]: I'm somewhat agnostic. I agree that `make` has many idiosyncrasies, but this code is pretty basic. Let me peek back at the `Dockerfile.py` approach and sleep on it and I'll ping back here.

**Comment by almussel** [comment]
Is there any reason to build our own spark distribution for newer versions that already have an updated jodatime version?

**Comment by fnothaft** [comment]
> Is there any reason to build our own spark distribution for newer versions that already have an updated jodatime version?

Oh! I didn't realize this was fixed in newer versions. Do you know when this was fixed?

**Comment by almussel** [comment]
@fnothaft It was fixed in 1.6 I believe

**Comment by fnothaft** [comment]
> @fnothaft It was fixed in 1.6 I believe

Ah, OK! Let me test this. If it was fixed in 1.6, then why don't we just move to 1.6.2 and remove the custom build cruft? ADAM's ToT targets Spark 1.6 by default, and also supports Spark 2.

**Comment by almussel** [comment]
@fnothaft That sounds great to me. I was under the impression we were stuck using 1.5 for ADAM...

**Comment by fnothaft** [comment]
@almussel oh no! ADAM runs on all Spark versions from 1.3.0 to 2.0.1. We actually run our [CI against every single one of these versions](https://amplab.cs.berkeley.edu/jenkins/job/ADAM/), just to check.

**Pull request closed without merging**